apps/analytics/views.py

```diff
@@ -205,3 +207,49 @@ def get_last_submission_datetime_analysis(request, challenge_pk, challenge_phase
     except:
         response_data = {'error': 'Bad request. Please try again later!'}
         return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+
+@throttle_classes([UserRateThrottle])
+@api_view(['GET', ])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((ExpiringTokenAuthentication,))
+def get_challenge_phase_submission_analysis(request, challenge_pk, challenge_phase_pk):
+    """
+    Returns
+    1. Total number of submissions in a challenge phase
+    2. Number of teams which made submissions in a challenge phase
+    3. Number of submissions with status a)Submitting, b)Submitted, c)Running, d)Failed, e)Cancelled, f)Finished status
+    4. Number of flagged & public submissions in challenge phase
+    """
+
+    challenge = get_challenge_model(challenge_pk)
+
+    challenge_phase = get_challenge_phase_model(challenge_phase_pk)
+
+    # Get the total submissions in a challenge phase
+    submissions = Submission.objects.filter(
+        challenge_phase=challenge_phase, challenge_phase__challenge=challenge)
+    total_submissions = submissions.count()
+
+    # Get the total participant teams in a challenge phase
+    participant_team_count = submissions.values('participant_team').distinct().count()
+
+    # Get flagged and public submission counts
+    flagged_submissions_count = submissions.filter(is_flagged=True).count()
+
+    public_submissions_count = submissions.filter(is_public=True).count()
+
+    flagged_and_public_submissions = {'is_flagged_count': flagged_submissions_count,
```

Comment:
Is there any specific reason for combining the flagged and public submissions? Both are totally different things.